src/psyclone/parse/module_manager.py

```diff
@@ -0,0 +1,324 @@
+                self._search_paths.append(directory)
+            if not recursive:
+                break
+            for root, dirs, _ in os.walk(directory):
+                for current_dir in dirs:
+                    new_dir = os.path.join(root, current_dir)
+                    if new_dir not in self._search_paths:
+                        self._search_paths.append(new_dir)
+
+    # ------------------------------------------------------------------------
+    def _add_all_files_from_dir(self, directory):
+        '''This function adds all files with an extension of (F/f/X/x)90 in the
+        given directory to the mapping of module names to file names. The
+        module names are based on the filename using `get_modules_in_file()`.
+        By default it is assumed that `a_mod.f90` contains the module `a_mod`.
+
+        :param str directory: the directory to list all files from
+        '''
+
+        with os.scandir(directory) as all_entries:
+            for entry in all_entries:
+                _, ext = os.path.splitext(entry.name)
+                if (not entry.is_file()) or \
+                        ext not in [".F90", ".f90", ".X90", ".x90"]:
+                    continue
+                full_path = os.path.join(directory, entry.name)
+                all_modules = self.get_modules_in_file(full_path)
+                for module in all_modules:
+                    if module not in self._mod_2_filename:
+                        mod_info = ModuleInfo(module, full_path)
+                        self._mod_2_filename[module] = mod_info
+
+    # ------------------------------------------------------------------------
+    def get_module_info(self, module_name):
+        '''This function returns the ModuleInformation for the specified
+        module.
+
+        :param str module_name: name of the module.
+
+        :returns: the filename that contains the module
+        :rtype: str
+
+        :raises FileNotFoundError: if the module_name is not found in \
+            either the cached data nor in the search path.
+
+        '''
+        mod_lower = module_name.lower()
+
+        # First check if we already know about this file:
+        mod_info = self._mod_2_filename.get(mod_lower, None)
+        if mod_info:
+            return mod_info
+
+        # If not, check the search paths. To avoid frequent accesses to
+        # the directories, we search directories one at a time, and
+        # add the list of all files in that directory to our cache
+        # _mod_2_filename
+        while self._search_paths:
+            # Get the first element from the search path list:
+            directory = self._search_paths.pop(0)
+            self._add_all_files_from_dir(directory)
+            mod_info = self._mod_2_filename.get(mod_lower, None)
+            if mod_info:
+                return mod_info
+
+        raise FileNotFoundError(f"Could not find source file for module "
+                                f"'{module_name}'.")
```

Comment:
That is a bit of a problem, since I don't have the search path anymore. Considering that the search paths are taken from the command line, the user should be able to see it. But if you prefer, I can of course store a copy of the original search paths and print it. Let me know.

Reply:
How about a compromise: you could extend the message to say that the location needs to be supplied via the (`-I`) command-line flags?

Reply:
You are right, I kicked my lazy a** and added the full search path and help how to add a path.